### PR TITLE
[docs] add notes for stateful model changes

### PR DESCRIPTION
**Pull request by williamberman**
re: https://github.com/huggingface/diffusers/issues/2907

Ideally we should fully fix this properly, but such issues with the statefulness of offloading don't seem to be that frequent so lets just document it for now

### CODE REVIEW

File: docs/source/en/optimization/fp16.mdx [hunk 1]
```diff
@@ -251,6 +253,10 @@ image = pipe(prompt).images[0]
 This feature requires `accelerate` version 0.17.0 or larger.
 </Tip>
 
+**Note**: `enable_model_cpu_offload()` is a stateful operation that installs hooks on the models and state on the pipeline. In order to properly re-offload
+models after they are called, it is required that the entire pipeline is run and models are called in the order the pipeline expects them to be. Exercise caution
+if models are re-used outside the context of the pipeline after hooks have been installed.
```
Comment (sayakpaul) [inline docs/source/en/optimization/fp16.mdx:258]: Could we also include notes on discarding the installed hooks?

File: docs/source/en/optimization/fp16.mdx [hunk 2]
```diff
@@ -251,6 +253,10 @@ image = pipe(prompt).images[0]
 This feature requires `accelerate` version 0.17.0 or larger.
 </Tip>
 
+**Note**: `enable_model_cpu_offload()` is a stateful operation that installs hooks on the models and state on the pipeline. In order to properly re-offload
```
Comment (pcuenca) [inline docs/source/en/optimization/fp16.mdx:256]:
```suggestion
**Note**: `enable_model_cpu_offload()` is a stateful operation that installs hooks on the models and state on the pipeline. In order to properly offload
```

(I might be mistaken, didn't understand what `re-offload` was referring to)

**Review by pcuenca — APPROVED** [review]

**Review by patrickvonplaten — APPROVED** [review]
Cool, we should also soon think about how we can remedy this somehow

File: docs/source/en/optimization/fp16.mdx (hunk 2 above)
Comment (williamberman) [reply docs/source/en/optimization/fp16.mdx:256]: re-offload meaning move the weights are moved back to the cpu

File: docs/source/en/optimization/fp16.mdx (hunk 1 above)
Comment (williamberman) [reply docs/source/en/optimization/fp16.mdx]: https://github.com/huggingface/diffusers/pull/3252/commits/c099a8fa5608231c991c5608ac4bf5df6b194b39

**Pull request merged**